code/game/machinery/adv_med.dm

```diff
@@ -355,46 +362,40 @@
-		data["dermAmt"]			= R.get_reagent_amount("dermaline")
-		data["otherAmt"]		= R.total_volume - (data["soporAmt"] + data["dexAmt"] + data["bicardAmt"] + data["norepiAmt"] + data["dermAmt"])
-		data["bodyparts"]		= get_external_wound_data(occupant)
-		data["organs"]			= get_internal_wound_data(occupant)
+		VUEUI_SET_CHECK(data["stat"], occupant.stat, ., data)
+		VUEUI_SET_CHECK(data["name"], occupant.name, ., data)
+		VUEUI_SET_CHECK(data["species"], occupant.get_species(), ., data)
+		VUEUI_SET_CHECK(data["brain_activity"], brain_result, ., data)
+		VUEUI_SET_CHECK(data["pulse"], text2num(pulse_result), ., data)
+		VUEUI_SET_CHECK(data["blood_pressure"], occupant.get_blood_pressure(), ., data)
+		VUEUI_SET_CHECK(data["blood_pressure_level"], occupant.get_blood_pressure_alert(), ., data)
+		VUEUI_SET_CHECK(data["blood_volume"], occupant.get_blood_volume(), ., data)
+		VUEUI_SET_CHECK(data["blood_o2"], occupant.get_blood_oxygenation(), ., data)
+		VUEUI_SET_CHECK(data["rads"], occupant.total_radiation, ., data)
+
+		VUEUI_SET_CHECK(data["cloneLoss"], get_severity(occupant.getCloneLoss()), ., data)
+		VUEUI_SET_CHECK(data["oxyLoss"], get_severity(occupant.getOxyLoss()), ., data)
+		VUEUI_SET_CHECK(data["bruteLoss"], get_severity(occupant.getBruteLoss()), ., data)
+		VUEUI_SET_CHECK(data["fireLoss"], get_severity(occupant.getFireLoss()), ., data)
+		VUEUI_SET_CHECK(data["toxLoss"], get_severity(occupant.getToxLoss()), ., data)
+
+		VUEUI_SET_CHECK(data["paralysis"], occupant.paralysis, ., data)
+		VUEUI_SET_CHECK(data["bodytemp"], occupant.bodytemperature, ., data)
+		VUEUI_SET_CHECK(data["occupant"], occupant, ., data)
+		VUEUI_SET_CHECK(data["norepiAmt"], R.get_reagent_amount("norepinephrine"), ., data)
+		VUEUI_SET_CHECK(data["soporAmt"], R.get_reagent_amount("stoxin"), ., data)
+		VUEUI_SET_CHECK(data["bicardAmt"], R.get_reagent_amount("bicaridine"), ., data)
+		VUEUI_SET_CHECK(data["dexAmt"], R.get_reagent_amount("dexalin"), ., data)
+		VUEUI_SET_CHECK(data["dermAmt"], R.get_reagent_amount("dermaline"), ., data)
+		VUEUI_SET_CHECK(data["otherAmt"], R.total_volume - (data["soporAmt"] + data["dexAmt"] + data["bicardAmt"] + data["norepiAmt"] + data["dermAmt"]), ., data)
+		VUEUI_SET_CHECK(data["bodyparts"], get_external_wound_data(occupant), ., data)
+		VUEUI_SET_CHECK(data["organs"], get_internal_wound_data(occupant), ., data)
 		var/list/missing 		= get_missing_organs(occupant)
-		data["missingparts"]	= missing
-		data["hasmissing"]		= missing.len ? 1 : 0
-		data["hasvirus"]		= occupant.virus2.len || occupant.viruses.len
-		data["hastgvirus"]		= occupant.viruses.len
-		data["tgvirus"]			= occupant.viruses
-
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
-	if (!ui)
-		ui = new(user, src, ui_key, "med_diagnostics.tmpl", "Medical Diagnostics", 800, 500, state = state)
-		ui.set_initial_data(data)
-		ui.open()
-		ui.set_auto_update(1)
+		VUEUI_SET_CHECK(data["missingparts"], missing, ., data)
```

Comment:
This is absolute flop. List instances always will differ it UI pushes change. You can't do this with set check.